.cirrus.yml

```diff
@@ -21,6 +23,29 @@ vm_instance_template: &VM_TEMPLATE
   region: eu-central-1
   subnet_id: ${CIRRUS_AWS_SUBNET}
 
+multi_arch_build_gcp_task: 
+  only_if: $CIRRUS_CRON == 'nightly-mend-scan'
+  env:
+    matrix:
+      - version: 10/datacenter/app
+        tag: $CURRENT_VERSION-datacenter-app
+      - version: 10/datacenter/search
+        tag: $CURRENT_VERSION-datacenter-search
+  ec2_instance:
+    <<: *VM_TEMPLATE
+  login_script:
+    - export DOCKER_GCLOUD_PASSWORD=$(echo ${DOCKER_GCLOUD_SA_KEY} | base64 -d)
+    - docker login -u _json_key -p "$DOCKER_GCLOUD_PASSWORD" https://${GCLOUD_REGISTRY}
+  setup_script:
+    - docker run -t --rm --privileged tonistiigi/binfmt --install all
+    - docker buildx create --name multibuilder --driver docker-container
+    - docker buildx use multibuilder
+    - docker buildx inspect --bootstrap
+  build_script:
+    - echo "Build the ${PUBLIC_IMAGE_NAME}:${tag} image supporting two architectures, linux/amd64 and linux/arm64 and push to the GCP registry ${GCLOUD_REGISTRY}"
+    - git fetch --tags
+    - git checkout "${CURRENT_VERSION}"
+    - .cirrus/multi-arch-build.sh ${GCLOUD_REGISTRY}/${PUBLIC_IMAGE_NAME} ${tag} ${version}
```

Comment:
Considering the docker hub automatically rebuild, could we just pull, tag and push bash to the new registry ? 
This would prevent to use CPU on building something that already exist, wdyt ? 

Also we need to be able to parameterize a bit differently for the GCP marketplace.

We will have registry/productname for the main image.
and registry/productname/searchImage for the search nodes.
and registry/productname/deployer for the deployer that will come a bit later on.

Wdyt ?

Reply:
We should also always tag both the track and minor, so for SQ 10.4.0 we should tag it for both 10.4 and 10.4.0

Reply:
Thanks for your comment. Could you please have a look now? I left the other tasks commented, so that you can easily rerun and check what we have in the registry.